main.tf

```diff
@@ -89,6 +107,22 @@ resource "aws_s3_bucket" "aws_logs" {
   }
 }
 
+resource "aws_s3_bucket_public_access_block" "public_access_block" {
```

Comment:
Consider splitting this work out into its own PR. It doesn't seem related to the PR topic.

Reply:
Adding it here actually brought up an edge case that will have to be dealt with, which is removing the s3 bucket policy (ensure-private-read-write for s3 resource) makes it possible for the template to be "empty", and the policy therefore to not be attached for an s3 logging bucket for s3. This was worked around by creating an `aws_s3_bucket_policy` and attaching it to the bucket, depending on whether the bucket is for all services, or a single service that's NOT s3.